pyproject.toml

```diff
@@ -134,6 +134,7 @@ rope = { version = "*", optional = true }  # groups = ["dev"]
 doc8 = { version = "*", optional = true }  # groups = ["dev"]
 readme-renderer = { version = "*", optional = true }  # groups = ["dev"]
 tabulate = { version = "*", optional = true }  # groups = ["dev", "doctests"]
+fsspec = "^0.8.7"
```

Comment:
Out of interest, what was this for?

Reply:
Streamlit wasn't wanting to play nice with the indent parameter of pandas.to_json. It would work outside of streamlit, but within the app I would get an error message saying I needed that package. 

Also, am I doing things wrong when adding to poetry? I just did: 
`poetry add <package>`

Just trying to make sure I don't mess anything up in the future

Reply:
> but within the app I would get an error message saying I needed that package

Would you be able to provide a print out of that error message? I've never seen that one before.

> Also, am I doing things wrong when adding to poetry?

For normal poetry usage that would be fine. But we are using it a little bit differently so as to support optional imports. Instead of using `poetry add` I would recommend instead just inserting a line within `pyproject.toml` that looks like:

```toml
fsspec = { version = "*", optional = true }  # groups = ["user", "dev"]
```

And then place it under the appropriate section under https://github.com/pymedphys/pymedphys/blob/4b983b84dad4cc5bb67ac673f273d9c77a2a11d2/pyproject.toml#L37-L39

Then run:

```
poetry run pymedphys dev propagate
poetry install -E dev
```

---

When you are adding dependencies though it is a bit of a big deal. So, I'd recommend trying to reserve the dependency addition for its own PR, and have a reasonable justification for the addition. Hence why I'm trying to be sure that we really do need `fsspec` and can't get around it some other way.

Reply:
![image](https://user-images.githubusercontent.com/40006991/111054541-2f7f4300-8433-11eb-9301-0eeae29655cb.png)

Reply:
Here is the error message I get when trying to run without fsspec. This only was happening when running in streamlit. I was able to write the dataframe to json outside of streamlit with no issues.

Reply:
Yup, awesome. Makes sense. That's a pandas optional dependency. Let's merge this PR and then could you make a fresh PR where you just add that new dependency by itself?